packages/lesswrong/components/ultraFeed/UltraFeedPostItem.tsx

```diff
@@ -147,33 +152,43 @@ const UltraFeedPostItem = ({
   showKarma,
   settings = DEFAULT_SETTINGS,
 }: {
-  post: UltraFeedPostFragment,
+  post: PostsListWithVotes,
   postMetaInfo: FeedPostMetaInfo,
   index: number,
   showKarma?: boolean,
   settings?: UltraFeedSettingsType,
 }) => {
   const classes = useStyles(styles);
-  const { PostActionsButton, FeedContentBody, UltraFeedItemFooter, FormatDate, Loading, TruncatedAuthorsList } = Components;
+  const { PostActionsButton, FeedContentBody, UltraFeedItemFooter, FormatDate, Loading, TruncatedAuthorsList, OverflowNavButtons } = Components;
 
   const { observe, trackExpansion } = useUltraFeedObserver();
   const elementRef = useRef<HTMLDivElement | null>(null);
   const metaLeftSectionRef = useRef<HTMLDivElement>(null);
+  const { openDialog } = useDialog();
+  const overflowNav = useOverflowNav(elementRef);
   const { captureEvent } = useTracking();
-  const { isAnon, authors } = usePostsUserAndCoauthors(post);
   const { recordPostView, isRead } = useRecordPostView(post);
   const [hasRecordedViewOnExpand, setHasRecordedViewOnExpand] = useState(false);
-  const [isLoadingFull, setIsLoadingFull] = useState(false);
+  const isForeignCrosspost = isPostWithForeignId(post) && !post.fmCrosspost.hostedHere
+  const [isLoadingFull, setIsLoadingFull] = useState(isForeignCrosspost);
   const [shouldShowLoading, setShouldShowLoading] = useState(false);
-  const [expandLevel, setExpandLevel] = useState(0);
+  const [resetSig, setResetSig] = useState(0);
+
+
+  const apolloClient = useForeignApolloClient();
+  
+  const documentId = isForeignCrosspost ? (post.fmCrosspost.foreignPostId ?? undefined) : post._id;
 
   const { document: fullPost, loading: loadingFullPost } = useSingle({
-    documentId: post._id,
+    documentId,
     collectionName: "Posts",
-    fragmentName: "PostsExpandedHighlight",
-    skip: !isLoadingFull,
+    apolloClient: isForeignCrosspost ? apolloClient : undefined,
+    fragmentName: isForeignCrosspost ? "SunshinePostsList" : "PostsExpandedHighlight",
```

Comment:
why a different fragment in the crosspost case?  (and a moderation fragment, no less)